staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -569,6 +571,83 @@ func (s *sharedIndexInformer) HandleDeltas(obj interface{}) error {
 	return nil
 }
 
+// EventHandlerRemovable is an extension interface
+// for a SharedIndexInformer that allows to remove event handlers again.
+// It is possible to remove only event handlers that are (go) comparable,
+// This means event handlers like ResourceEventHandlerFuncs cannot be removed
+// if added by value and not by reference, because they contain function pointers
+// that are not comparable in go. Trying to remove an uncomparable handler
+// results in an appropriate error.
+// This feature can be used to dynamically add and remove watches again depending
+// on the actual state of a controller.
+type EventHandlerRemovable interface {
+	RemoveEventHandler(handler ResourceEventHandler) error
+}
+
+// ExtendedSharedIndexInformer is an extended interface to the
+// SharedIndexInformer informer interface implemented by the standard
+// sharedIndexInformer implementation,
+// It offers the removing of handlers and some state info useful
+// to use after handler removal to decide whether an informer
+// is still required.
+type ExtendedSharedIndexInformer interface {
+	SharedIndexInformer
+	EventHandlerRemovable
+	HasEventHandlers() bool
+	IsStopped() bool
+	IsStarted() bool
+}
+
+// IsStarted reports whether the informaer has already been started
+func (s *sharedIndexInformer) IsStarted() bool {
+	s.startedLock.Lock()
+	defer s.startedLock.Unlock()
+	return s.started
+}
+
+// IsStopped reports whether the informaer has already been stopped
+func (s *sharedIndexInformer) IsStopped() bool {
+	s.startedLock.Lock()
+	defer s.startedLock.Unlock()
+	return s.stopped
+}
+
+// HasEventHandlers reports whether the informer still has registered
+// event handlers
+func (s *sharedIndexInformer) HasEventHandlers() bool {
```

Comment:
For the functionality o managing informers this would not be necessary, but you are right, for debugging it might be useful.
I changed the name and semantics accordingly